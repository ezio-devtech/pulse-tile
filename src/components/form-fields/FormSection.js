import React, { PureComponent } from 'react';
// import PropTypes from 'prop-types';
import classNames from 'classnames';
// import _ from 'lodash/fp';

export default class FormSecrion extends PureComponent {
  // static propTypes = {
  //   label: PropTypes.string.isRequired,
  //   input: PropTypes.object.isRequired,
  //   meta: PropTypes.shape({
  //     active: PropTypes.bool,
  //     error: PropTypes.any,
  //   }).isRequired,
  //   id: PropTypes.string.isRequired,
  // };
  // getTheme() {
  //   const { isBordered, theme, children } = this.props;
  // }

  render() {
    const { title, isImportBtn, isAccordion, isBordered, theme, children } = this.props;

    // const theme = this.getTheme();

    return (
      <div className={classNames('form-group-section', { theme: theme }, { 'form-group-section-bordered': isBordered }, { 'accordion': isAccordion })}>
        <div className="form-group-section-body">
          {children}
        </div>
      </div>
    //   <div className="form-group-section form-group-section-bordered form-group-section-primary accordion" key={index}>
    //   <div className="form-group-section-heading">
    //   <div className="control-group without-side-indent right">
    //   {/*<button className="btn btn-primary" ng-click="importToCreate('medications', medicationItem)"><span className="btn-text">Import Data</span></button>*/}
    // <button className="btn btn-primary"><span className="btn-text">Import Data</span></button>
    // {/*<button className="btn btn-primary btn-inverse btn-square btn-form-group-section-toggle" ng-click="toggleSubAccordion()"><i className="btn-icon fa fa-chevron-up"></i></button>*/}
    // <button className="btn btn-primary btn-inverse btn-square btn-form-group-section-toggle"><i className="btn-icon fa fa-chevron-up"></i></button>
    // </div>
    // <h3 className="panel-title"></h3>
    // </div>
    // <div className="form-group-section-body">
    //
    // </div>
    )
  }
}

