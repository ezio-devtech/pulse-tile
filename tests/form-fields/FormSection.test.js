import React from 'react';
import {configure, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-15';
configure({ adapter: new Adapter() });

import FormSection from '../../src/components/form-fields/FormSection';

const testProps = {
  onImportClick: () => {},
  title: 'Test title',
  theme: 'primary',
};
// isImportBtn: ,
// isAccordion: ,
// isBordered: ,

describe('Component <FormSectionList />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
      <FormSection
        onImportClick={testProps.onImportClick}
        title={testProps.title}
        theme={testProps.theme}
      >
        <span className='FormSection-children'>FormSectionList children</span>
      </FormSection>);

    expect(component.instance().props['onImportClick']).toEqual(testProps.onImportClick);
    expect(component.instance().props['title']).toEqual(testProps.title);
    expect(component.instance().props['theme']).toEqual(testProps.theme);
    expect(component.instance().props['isImportBtn']).toEqual(false);
    expect(component.instance().props['isAccordion']).toEqual(false);
    expect(component.instance().props['isBordered']).toEqual(false);


    expect(component.find('.form-group-section')).toHaveLength(1);
    expect(component.find(`.form-group-section-${testProps.theme}`)).toHaveLength(1);
    expect(component.find('.form-group-section-bordered')).toHaveLength(0);
    expect(component.find('.accordion')).toHaveLength(0);
    expect(component.find('.accordion.open')).toHaveLength(0);
    expect(component.find('.form-group-section-heading')).toHaveLength(1);
    expect(component.find('.control-group')).toHaveLength(0);
    expect(component.find('.panel-title')).toHaveLength(1);
    expect(component.find('.panel-title').text()).toEqual(testProps.title);
    expect(component.find('.FormSection-children')).toHaveLength(1);

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  it('should renders without title', () => {
    const component = shallow(
      <FormSection
        onImportClick={testProps.onImportClick}
        theme={testProps.theme}
      >
        <span className='FormSection-children'>FormSectionList children</span>
      </FormSection>);

    expect(component.instance().props['title']).toEqual(undefined);

    expect(component.find('.form-group-section-heading')).toHaveLength(0);
    expect(component.find('.panel-title')).toHaveLength(0);

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});