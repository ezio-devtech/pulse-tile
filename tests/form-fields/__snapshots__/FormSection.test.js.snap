// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <FormSectionList /> should renders with all props correctly 1`] = `
<div
  className="form-group-section form-group-section-primary"
>
  <div
    className="form-group-section-heading"
  >
    <h3
      className="panel-title"
    >
      Test title
    </h3>
  </div>
  <div
    className="form-group-section-body"
  >
    <span
      className="FormSection-children"
    >
      FormSectionList children
    </span>
  </div>
</div>
`;

exports[`Component <FormSectionList /> should renders without title 1`] = `
<div
  className="form-group-section form-group-section-primary"
>
  <div
    className="form-group-section-body"
  >
    <span
      className="FormSection-children"
    >
      FormSectionList children
    </span>
  </div>
</div>
`;
